### ISSUE TITLE
artifacts with very large bilinear patch

### ISSUE THREAD

**Issue by mmp** [mmp/pbrt-v4#186]
(Maybe an issue with precision and surface normals??)

Rendering
```
Sampler "halton" 
    "integer pixelsamples" 2048

Film "rgb" "float iso" 200
    "integer yresolution" [ 800 ]
    "integer xresolution" [ 1200 ]
 
Scale -1 1 1
LookAt 3.69558 -3.46243 1.7
    3.04072 -2.85176 1.5
    -0.317366 0.312466 0.895346

Camera "perspective"
    "float fov" [ 38 ]
#    "float focaldistance" 4.21 "float lensradius" .15

Integrator "volpath" "integer maxdepth" 1

WorldBegin

AttributeBegin
    LightSource "infinite"
AttributeEnd

AttributeBegin
    Material "coateddiffuse" "float roughness" .1
        "rgb reflectance" [ .2 .2 .2 ]
    Shape "bilinearmesh"
    "point3 P" [ -10000 -10000 0  10000 -10000 0   -10000 10000 0  10000 10000 0 ]
AttributeEnd
```

Gives:
<img width="958" alt="Screen Shot 2021-09-24 at 10 08 27 AM" src="https://user-images.githubusercontent.com/102567/134714244-d34181d9-0b7c-4d09-8e8b-aad3e69836f4.png">

A 100x100 patch is fine:
<img width="920" alt="Screen Shot 2021-09-24 at 10 09 26 AM" src="https://user-images.githubusercontent.com/102567/134714340-dd879558-aa59-407b-a8c1-c23a1d045f59.png">

**Comment by shadeops** [comment]
Apologies for creeping issues you submitted for yourself, but I noticed that this is now fixed courtesy of b03a013b and 6e519a5c

Looks fine even with
```
 "point3 P" [ -1000000 -1000000 0  1000000 -1000000 0   -1000000 1000000 0  1000000 1000000 0 ]
```

Yay?

**Comment by mmp** [comment]
Not at all creeping! Great to know that other fix happened to take care of this.
